model/ListElement/Service/ListElementsFinder.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoBackOffice\model\ListElement\Service;
+
+use core_kernel_classes_Class;
+use oat\tao\model\Context\ContextInterface;
+use oat\tao\model\Lists\Business\Domain\ValueCollection;
+use oat\tao\model\Lists\Business\Service\ValueCollectionService;
+use oat\tao\model\Lists\Business\Input\ValueCollectionSearchInput;
+use oat\tao\model\Lists\Business\Domain\ValueCollectionSearchRequest;
+use oat\taoBackOffice\model\ListElement\Context\ListElementsFinderContext;
+use oat\taoBackOffice\model\ListElement\Contract\ListElementsFinderInterface;
+
+class ListElementsFinder implements ListElementsFinderInterface
+{
+    /** @var ValueCollectionService */
+    private $valueCollectionService;
+
+    public function __construct(ValueCollectionService $valueCollectionService)
+    {
+        $this->valueCollectionService = $valueCollectionService;
+    }
+
+    public function find(ContextInterface $context): ValueCollection
+    {
+        $request = new ValueCollectionSearchRequest();
+
+        /** @var core_kernel_classes_Class $listClass */
+        $listClass = $context->getParameter(ListElementsFinderContext::PARAMETER_LIST_CLASS);
+        $request->setValueCollectionUri($listClass->getUri());
+
+        $totalCount = $this->valueCollectionService->count(new ValueCollectionSearchInput($request));
```

Comment:
Weren't you adding this to inside `$this->valueCollectionService->findAll` method to populate `valueCollection`?

Reply:
I'm still not sure, that this is good idea - combine 2 actions in a single method (as we have 2 methods: 1 to receive values and 1 to count).
If we will combine this - we will do this action (counting) every time without possibility to skip this)

Reply:
You could have, as doctrine does, a `callable` inside the `ValueCollection` that counts the total for you only when you call the method `getTotal`.

But it is optional, I am thinking about usability.

Reply:
I will think today about this 🙂